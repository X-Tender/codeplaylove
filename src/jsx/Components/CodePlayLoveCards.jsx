import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import convert from 'htmr';
import { getCards } from 'reducers/cards';
import Section from 'Semantics/Section';
import P from 'Semantics/P';
import A from 'Semantics/A';

const CodePlayLoveCard = ({ title, children, style }) => (
	<li className={`code-play-love-card code-play-love-card--${style}`}>
		<div className={`code-play-love-card__icon code-play-love-card__icon--${style}`} />
		<div className="code-play-love-card__content">
			<h3 className="code-play-love-card__head">{title}</h3>
			{children}
		</div>
	</li>
);

class CodePlayLoveCards extends PureComponent {
	constructor(props) {
		super(props);

		if (this.props.cards.data.length === 0) this.props.getCards();
	}

	render() {
		const cardsData = this.props.cards.data;

		const cards = cardsData.sort((a, b) => a.sort - b.sort).map(({ id, title, copy, style }) => (
			<CodePlayLoveCard key={id} title={title} style={style}>
				{copy &&
					convert(copy, {
						transform: {
							p: P,
							a: A,
						},
					})}
			</CodePlayLoveCard>
		));

		return (
			<Section className="code-play-love-cards">
				<ul className="code-play-love-cards__list">{cards}</ul>
			</Section>
		);
	}
}

const mapStateToProps = ({ cards }) => ({
	cards,
});

export default connect(
	mapStateToProps,
	{ getCards }
)(CodePlayLoveCards);
