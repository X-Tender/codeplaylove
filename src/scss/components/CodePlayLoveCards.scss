.code-play-love-cards {
	background-color: $secondaryBackgroundColor;
	padding-top: 50px;
	padding-bottom: 50px;
	&__list {
		grid-column: content;
		display: grid;
		grid-gap: 25px;
		grid-template-columns: 1fr;
		justify-content: center;

		@include sm {
			grid-template-columns: repeat(3, minmax(0, calc((1200px - (3 * 25px)) / 4)));
		}
	}
}

.code-play-love-card {
	color: #fff;
	display: flex;
	flex-direction: column;
	align-items: center;
	position: relative;
	padding: 30px;
	padding-bottom: 40px;
	border-radius: $globalBorderRadius;
	transition: transform $defaultTransitionDuration ease;
	z-index: 1;
	will-change: transition;

	&:hover {
		transform: scale(1.1);
		z-index: 2;
	}

	&--code {
		background-color: $codeColor;
		box-shadow: 0 5px 30px rgba($codeColor, $glowShadowAlpha);
	}

	&--play {
		background-color: $playColor;
		box-shadow: 0 5px 30px rgba($playColor, $glowShadowAlpha);
	}

	&--love {
		background-color: $loveColor;
		box-shadow: 0 5px 30px rgba($loveColor, $glowShadowAlpha);
	}

	&__icon {
		size: 80px;

		&--code {
			background: svg-load('icon_code.svg', fill=#fff);
		}

		&--play {
			background: svg-load('icon_play.svg', fill=#fff);
		}

		&--love {
			background: svg-load('icon_love.svg', fill=#fff);
		}
	}

	&__content {
		margin-top: 20px;

		p {
			font-size: 17px;
			hyphens: auto;
		}
	}

	&__head {
		font-family: 'Raleway', sans-serif;
		text-transform: uppercase;
		font-size: 18px;
		font-weight: 700;
		text-align: center;
		margin-bottom: 20px;
	}
}
