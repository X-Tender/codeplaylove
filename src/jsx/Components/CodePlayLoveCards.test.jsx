import React from 'react';
import { shallow, mount } from 'enzyme';
import { CodePlayLoveCards } from './CodePlayLoveCards';

describe('CodePlayLoveCards', () => {
	const getCards = jest.fn();

	const cards = {
		loaded: false,
		data: [],
	};

	const props = {
		getCards,
		cards,
	};
	let cplCards = shallow(<CodePlayLoveCards {...props} />);

	it('renders correctly', () => {
		expect(cplCards).toMatchSnapshot();
	});

	describe('when mounted', () => {
		beforeEach(() => {
			cplCards = mount(<CodePlayLoveCards {...props} />);
		});

		it('dispatches the `getCards()` method when mounted and no cards loaded', () => {
			expect(getCards).toHaveBeenCalled();
		});
	});
});
