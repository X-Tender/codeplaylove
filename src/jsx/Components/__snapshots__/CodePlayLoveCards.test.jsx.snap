// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodePlayLoveCards renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CodePlayLoveCards
    cards={
      Object {
        "data": Array [],
        "loaded": false,
      }
    }
    getCards={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ul
        className="code-play-love-cards__list"
      />,
      "className": "code-play-love-cards",
      "isFullWidth": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "code-play-love-cards__list",
      },
      "ref": null,
      "rendered": Array [],
      "type": "ul",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ul
          className="code-play-love-cards__list"
        />,
        "className": "code-play-love-cards",
        "isFullWidth": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "code-play-love-cards__list",
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
